playbooks/roles/filebeat/tasks/main.yml

```diff
@@ -0,0 +1,28 @@
+- name: Ensure {{ filebeat_app_dir }} exists
+  file: path={{ filebeat_app_dir }} state=directory owner=root group=root mode=0777
+
+- name: ensure we have the specified filebeat release
+  get_url: url={{ filebeat_url }} dest=/tmp/{{ filebeat_file }}
+
+- name: Install filebeat
+  shell: dpkg --skip-same-version -i /tmp/{{ filebeat_file }}
+  register: dpkg_result
+
+- name: copy config
+  template:
+    src: filebeat.yml.j2
+    dest: "/etc/filebeat/filebeat.yml"
+    mode: 0744
+
+- name: create symlink to {{ filebeat_app_dir }} for config
+  file: src="/etc/filebeat/filebeat.yml" dest="{{ filebeat_app_dir }}/filebeat.yml" state=link
+
+- name: allow filebeat to access logs
+  file: 
+    path: /edx/var/log
+    mode: 0777
```

Comment:
This isn't correct;
it grants read-write access to _everyone_ on this server; that's surely not what we want here.

In order to determine how best to handle permissions, let's ask the following:
- Which task needs access?
- What username invokes that task?
- What files do they need access to?
- Do they really need write access? Is read-only enough?

Following the principle of least privilege, we should never grant more access than required [1].

[1] https://en.wikipedia.org/wiki/Principle_of_least_privilege

> In information security, computer science, and other fields, the principle of least privilege (also known as the principle of minimal privilege or the principle of least authority) requires that in a particular abstraction layer of a computing environment, every module (such as a process, a user, or a program, depending on the subject) must be able to access only the information and resources that are necessary for its legitimate purpose.